crates/migrate/src/lib.rs

```diff
@@ -75,16 +111,76 @@ pub async fn run(args: Args) -> anyhow::Result<()> {
                 tracing::error!(?err, "error subscribing to shutdown signal");
             }
         }
-    };
-    */
+    });
+
+    while !shutdown.is_finished() {
+        let migrations = sqlx::query!(
+            r#"
+            SELECT
+                m.id AS "id: models::Id",
+                m.catalog_name_or_prefix,
+                s.data_plane_name AS "src_plane",
+                t.data_plane_name AS "tgt_plane"
+            FROM data_plane_migrations m, data_planes s, data_planes t
+            WHERE m.active
+            AND m.cordon_at <= NOW()
+            AND m.src_plane_id = s.id
+            AND m.tgt_plane_id = t.id
+            "#
+        )
+        .fetch_all(pg_pool)
+        .await
+        .context("fetching pending migrations")?;
+
+        for migration in migrations {
+            tracing::info!(
+                id = ?migration.id,
+                catalog_name_or_prefix = %migration.catalog_name_or_prefix,
+                src_plane = ?migration.src_plane,
+                tgt_plane = ?migration.tgt_plane,
+                "performing scheduled migration"
+            );
+
+            () = migrate_data_planes(
+                &pg_pool,
+                &migration.src_plane,
+                &migration.tgt_plane,
+                &migration.catalog_name_or_prefix,
+            )
+            .await
+            .with_context(|| format!("failed to perform migration {}", migration.id))?;
+
+            sqlx::query!(
+                r#"
+                UPDATE data_plane_migrations
+                SET active = false
```

Comment:
So once the migration is done, by setting the migration to inactive, the cordon is gone?